core/domain/feedback_services.py

```diff
@@ -368,7 +421,123 @@ def _get_thread_from_model(thread_model):
         thread_model.id, thread_model.exploration_id, thread_model.state_name,
         thread_model.original_author_id, thread_model.status,
         thread_model.subject, thread_model.summary, thread_model.has_suggestion,
-        thread_model.created_on, thread_model.last_updated)
+        thread_model.message_count, thread_model.created_on,
+        thread_model.last_updated)
+
+
+def set_message_count_for_thread(full_thread_id, message_count):
+    """Assigns the message count field the number of messages present in the
+    thread.
+
+    Args:
+        full_thread_id: str. The full id of the thread.
+        message_count: int. The number of messages in the thread.
+    """
+    thread_model = feedback_models.FeedbackThreadModel.get(full_thread_id)
+    thread_model.message_count = message_count
+    thread_model.put()
+
+
+def increase_message_count_by_one(full_thread_id):
+    """Assigns the message count field the number of messages present in the
+    thread.
+
+    Args:
+        full_thread_id: str. The full id of the thread.
+        message_count: int. The number of messages in the thread.
+    """
+    thread_model = feedback_models.FeedbackThreadModel.get(full_thread_id)
+    thread_model.message_count += 1
+    thread_model.put()
+
+
+def get_thread_summaries(user_id, full_thread_ids):
+    """Returns a list of summaries corresponding to each of the threads given.
+
+    Args:
+        thread_ids: list(str). The ids of the threads excluding the exploration
+            id part.
+        exploration_ids: list(str). The ids of the explorations to which the
+            threads belong.
+
+    Returns:
+        list(dict). A list of dictionary containing the summaries of the threads
+            given to it.
+    """
+    exploration_and_thread_ids = (
+        [thread_id.split('.') for thread_id in full_thread_ids])
+    exploration_ids, thread_ids = zip(*exploration_and_thread_ids)
+
+    thread_model_ids = (
+        [feedback_models.FeedbackThreadModel.generate_full_thread_id(
+            exploration_id, thread_id)
+         for exploration_id, thread_id in zip(exploration_ids, thread_ids)])
+
+    feedback_thread_user_model_ids = (
+        [feedback_models.FeedbackThreadUserModel.generate_full_id(
+            user_id, exploration_id, thread_id)
+         for exploration_id, thread_id in zip(exploration_ids, thread_ids)])
+
+    last_two_messages_ids = (
```

Comment:
You're going to retrieve the thread models anyway. Why not retrieve them and convert to domain objects after generating thread_model_ids (above), and then obtain the last_two_messages_ids directly from there (you can define a helper function in the FeedbackThread domain object)? Then you don't need to get the thread models twice.